tests/integration/test_spans.py

```diff
@@ -1594,3 +1594,125 @@ def test_span_extraction_with_tags(
+                        "start_timestamp": start.timestamp(),
+                        "timestamp": end.timestamp() + 1,
+                        "exclusive_time": 345.0,  # The SDK knows that this span has a lower exclusive time
+                        "trace_id": "ff62a8b040f340bda5d830223def1d81",
+                        "data": {"sentry.release": "1.0"},
+                    },
+                ).encode()
+            ),
+        )
+    )
+
+    relay.send_envelope(project_id, envelope)
+
+    spans = spans_consumer.get_spans(timeout=10.0, max_attempts=6)
+    assert len(spans) == 0
+
+    def summarize_outcomes():
+        counter = Counter()
+        for outcome in outcomes_consumer.get_outcomes():
+            counter[(outcome["category"], outcome["outcome"])] += outcome["quantity"]
+        return counter
+
+    assert summarize_outcomes() == {(12, 1): 1}
+
+    spans_consumer.assert_empty()
+    outcomes_consumer.assert_empty()
+
+
+def test_span_filtering_with_releases_inbound_filter(
+    mini_sentry, relay_with_processing, spans_consumer, outcomes_consumer
+):
+    relay = relay_with_processing(options=TEST_CONFIG)
+    project_id = 42
+    project_config = mini_sentry.add_full_project_config(project_id)
+    project_config["config"]["features"] = ["organizations:standalone-span-ingestion"]
+    project_config["config"]["filterSettings"] = {"releases": {"releases": ["1.0"]}}
+
+    spans_consumer = spans_consumer()
+    outcomes_consumer = outcomes_consumer()
+
+    end = datetime.now(timezone.utc) - timedelta(seconds=1)
+    duration = timedelta(milliseconds=500)
+    start = end - duration
+    envelope = Envelope()
+    envelope.add_item(
+        Item(
+            type="span",
+            payload=PayloadRef(
+                bytes=json.dumps(
+                    {
+                        "description": "organizations/metrics/data",
+                        "op": "default",
+                        "span_id": "cd429c44b67a3eb1",
+                        "segment_id": "968cff94913ebb07",
+                        "start_timestamp": start.timestamp(),
+                        "timestamp": end.timestamp() + 1,
+                        "exclusive_time": 345.0,  # The SDK knows that this span has a lower exclusive time
+                        "trace_id": "ff62a8b040f340bda5d830223def1d81",
+                        "data": {"sentry.release": "1.0"},
+                    },
+                ).encode()
+            ),
+        )
+    )
+
+    relay.send_envelope(project_id, envelope)
+
+    spans = spans_consumer.get_spans(timeout=10.0, max_attempts=6)
+    assert len(spans) == 0
+
+    def summarize_outcomes():
+        counter = Counter()
+        for outcome in outcomes_consumer.get_outcomes():
+            counter[(outcome["category"], outcome["outcome"])] += outcome["quantity"]
+        return counter
+
+    assert summarize_outcomes() == {(12, 1): 1}
+
+    spans_consumer.assert_empty()
+    outcomes_consumer.assert_empty()
```

Comment:
nit: Because integration tests take a long time to run, I suggest keeping only one of these two integration tests, and cover other filters with unit tests. See for example https://github.com/getsentry/relay/blob/b5fa1ff0b90e04934c0ac1f09e6410bd70ce661b/relay-filter/src/csp.rs#L313-L326